src/http/mod.rs

```diff
@@ -0,0 +1,116 @@
+//! The HTTP client may vary depending on which one the user configures. This
+//! module contains the required logic to use different clients
+//! interchangeably.
+
+#[cfg(feature = "client-reqwest")]
+mod reqwest;
+#[cfg(feature = "client-ureq")]
+mod ureq;
+
+use crate::client::{ClientResult, Spotify};
+
+use std::collections::HashMap;
+
+use maybe_async::maybe_async;
+use serde_json::Value;
+
+pub type Headers = HashMap<String, String>;
+pub type Query = HashMap<String, String>;
+pub type Form = HashMap<String, String>;
+
+pub mod headers {
+    use crate::oauth2::Token;
+
+    // Common headers as constants
+    pub const CLIENT_ID: &str = "client_id";
+    pub const CODE: &str = "code";
+    pub const GRANT_AUTH_CODE: &str = "authorization_code";
+    pub const GRANT_CLIENT_CREDS: &str = "client_credentials";
+    pub const GRANT_REFRESH_TOKEN: &str = "refresh_token";
+    pub const GRANT_TYPE: &str = "grant_type";
+    pub const REDIRECT_URI: &str = "redirect_uri";
+    pub const REFRESH_TOKEN: &str = "refresh_token";
+    pub const RESPONSE_CODE: &str = "code";
+    pub const RESPONSE_TYPE: &str = "response_type";
+    pub const SCOPE: &str = "scope";
+    pub const SHOW_DIALOG: &str = "state";
```

Comment:
Is it a problem? SHOW_DIALOG should be `show_dialog`?

Reply:
Whoops! Fixed.